source/initial_conditions/interface.cc

```diff
@@ -85,68 +91,127 @@ namespace aspect
 
 
     template <int dim>
-    Interface<dim> *
-    create_initial_conditions (ParameterHandler &prm)
+    void
+    Manager<dim>::parse_parameters (ParameterHandler &prm)
     {
-      std::string model_name;
+      // find out which plugins are requested and the various other
+      // parameters we declare here
       prm.enter_subsection ("Initial conditions");
       {
-        model_name = prm.get ("Model name");
+        model_names
+          = Utilities::split_string_list(prm.get("List of model names"));
+
+        AssertThrow(Utilities::has_unique_entries(model_names),
+                    ExcMessage("The list of strings for the parameter "
+                               "'Initial conditions/List of model names' contains entries more than once. "
+                               "This is not allowed. Please check your parameter file."));
+
+        const std::string model_name = prm.get ("Model name");
+
+        AssertThrow (model_name == "unspecified" || model_names.size() == 0,
```

Comment:
Nice way to make it backward compatible. I think we should open github issues for all places where we have this kind of compatibility code to remind ourselves that we want to remove them again post-2.0. Can you do that?